### PR TITLE
Import pharmacies

### LINKED ISSUE #16: Import pharmacies
https://www.bda.bg/images/stories/documents/registers/Register_Apteki/20200224_Register_apteki.xls

Only import identifier, name and address (update the entity accordingly), we don't want to have duplicate data as it will likely get out of sync sooner than later

Transform them to SQL and insert them either using hibernate's initial data insertion methods or via the database migration tool

### PR DESCRIPTION
**Pull request by hovanessyan**

### CODE REVIEW

File: src/main/java/bg/ehealth/prescriptions/config/gateways/BdaGatewayProperties.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    @NotBlank
+    private final String host;
+    @NotBlank
+    private final String registersUrl;
+    @NotBlank
+    private final String registerPharmacies;
+    @DurationUnit(ChronoUnit.SECONDS)
+    private final Duration connectionTimeout;
+
+    public BdaGatewayProperties(String host, String registersUrl, String registerPharmacies,
```
Comment (Glamdring) [inline src/main/java/bg/ehealth/prescriptions/config/gateways/BdaGatewayProperties.java]: let's avoid constructor injection (e.g. it makes it difficult to use mocks).
I think plain old @Value("${..}" annotations (or if there's a non-constructor option for such binding).
Another good aspect of @Value is that in the future people will be able to find the properies by just searching through the project.

File: src/main/java/bg/ehealth/prescriptions/persistence/model/PharmacyView.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package bg.ehealth.prescriptions.persistence.model;
+
+public interface PharmacyView {
```
Comment (Glamdring) [inline src/main/java/bg/ehealth/prescriptions/persistence/model/PharmacyView.java:3]: I think we could do without projections in general. It's fine to stay here if you think, though.

File: src/test/java/bg/ehealth/prescriptions/services/PharmacyServiceImportStartIT.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class PharmacyServiceImportStartIT {
+
+    @Autowired
+    private PharmacyService service;
+
+    @Test
+    public void importPharmacyRegistry() {
+        service.importPharmacies();
```
Comment (Glamdring) [inline src/test/java/bg/ehealth/prescriptions/services/PharmacyServiceImportStartIT.java:19]: in addition to the test, Let's have a regular job that fetches data from the register and updates the prod database. Cron expression configurable, e.g. once a week.

File: src/main/java/bg/ehealth/prescriptions/rest/PharmaciesController.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    private final PharmacyService service;
+
+    public PharmaciesController(PharmacyService service) {
+        this.service = service;
+    }
+
+    @GetMapping("/pharmacies")
+    public ResponseEntity<List<PharmacyView>> getAllPharmacies() {
+        return ResponseEntity.ok(service.allPharmacies());
```
Comment (Glamdring) [inline src/main/java/bg/ehealth/prescriptions/rest/PharmaciesController.java]: just List<PharmacyView> would suffice.

Comment (hovanessyan) [reply src/main/java/bg/ehealth/prescriptions/rest/PharmaciesController.java]: all of the above are completed.

**Review by Glamdring — APPROVED** [review]

**Pull request merged**